src/System.Runtime/tests/System/String.cs

```diff
@@ -1273,6 +1273,37 @@ public static void TestToLowerToUpper()
     }
 
     [Fact]
+    public static void TestToLowerToUpperInvariant_ASCII()
+    {
+        char[] asciiChars = new char[128];
+        char[] asciiCharsUpper = new char[128];
+        char[] asciiCharsLower = new char[128];
+
+        for (int i = 0; i < asciiChars.Length; i++)
+        {
+            asciiChars[i] = asciiCharsUpper[i] = asciiCharsLower[i] = (char)i;
```

Comment:
Why assign asciiCharsUpper/Lower[i] here and then reassign it below?  What about something like:
```C#
for (int i = 0; i < asciiChars.Length; i++)  
{
    char c = (char)i;
    asciiChars[i] = c;  

    // Purposefully avoiding char.ToUpper/ToLower here so as not  
    // to use the same thing we're testing.  
    asciiCharsLower[i] = (c >= 'A' && c <= 'Z') ? (char)(c - 'A' + 'a') : c;
    asciiCharsUpper[i] = (c >= 'a' && c <= 'z') ? (char)(c - 'a' + 'A') : c;
}
```